File: src/j/implementation/microSet/InnerSet_10.java
package j.implementation.microSet;

import java.util.Collection;

public class InnerSet_10 <T> implements InnerSet<T>
{
        protected T element_1;
        protected T element_2;
        protected T element_3;
        protected T element_4;
        protected T element_5;
        protected T element_6;
        protected T element_7;
        protected T element_8;
        protected T element_9;
        protected T element_10;
        
    public InnerSet_10(T p_element_1, T p_element_2, T p_element_3, T p_element_4, T p_element_5, T p_element_6, T p_element_7, T p_element_8, T p_element_9, T p_element_10)
    {
                element_1 = p_element_1;
                element_2 = p_element_2;
                element_3 = p_element_3;
                element_4 = p_element_4;
                element_5 = p_element_5;
                element_6 = p_element_6;
                element_7 = p_element_7;
                element_8 = p_element_8;
                element_9 = p_element_9;
                element_10 = p_element_10;
            }
    
    @Override
    public InnerSet<T> addElement(T p_element)
    {
        if(element_1.equals(p_element) || element_2.equals(p_element) || element_3.equals(p_element) || element_4.equals(p_element) || element_5.equals(p_element) || element_6.equals(p_element) || element_7.equals(p_element) || element_8.equals(p_element) || element_9.equals(p_element) || element_10.equals(p_element))
            return this;
        else
            return new InnerSet_11(element_1, element_2, element_3, element_4, element_5, element_6, element_7, element_8, element_9, element_10 , p_element);
    }

    @Override
    @SuppressWarnings({ "rawtypes", "unchecked" })
    public InnerSet<T> removeElement(Object p_element)
    {
                if(element_1.equals(p_element))
            return new InnerSet_9(element_2, element_3, element_4, element_5, element_6, element_7, element_8, element_9, element_10);
         else                 if(element_2.equals(p_element))
            return new InnerSet_9(element_1, element_3, element_4, element_5, element_6, element_7, element_8, element_9, element_10);
         else                 if(element_3.equals(p_element))
            return new InnerSet_9(element_1, element_2, element_4, element_5, element_6, element_7, element_8, element_9, element_10);
         else                 if(element_4.equals(p_element))
            return new InnerSet_9(element_1, element_2, element_3, element_5, element_6, element_7, element_8, element_9, element_10);
         else                 if(element_5.equals(p_element))
            return new InnerSet_9(element_1, element_2, element_3, element_4, element_6, element_7, element_8, element_9, element_10);
         else                 if(element_6.equals(p_element))
            return new InnerSet_9(element_1, element_2, element_3, element_4, element_5, element_7, element_8, element_9, element_10);
         else                 if(element_7.equals(p_element))
            return new InnerSet_9(element_1, element_2, element_3, element_4, element_5, element_6, element_8, element_9, element_10);
         else                 if(element_8.equals(p_element))
            return new InnerSet_9(element_1, element_2, element_3, element_4, element_5, element_6, element_7, element_9, element_10);
         else                 if(element_9.equals(p_element))
            return new InnerSet_9(element_1, element_2, element_3, element_4, element_5, element_6, element_7, element_8, element_10);
         else                 if(element_10.equals(p_element))
            return new InnerSet_9(element_1, element_2, element_3, element_4, element_5, element_6, element_7, element_8, element_9);
                        else
            return this;
    }

    @Override
    public boolean containsElement(Object p_element)
    {
        return element_1.equals(p_element) || element_2.equals(p_element) || element_3.equals(p_element) || element_4.equals(p_element) || element_5.equals(p_element) || element_6.equals(p_element) || element_7.equals(p_element) || element_8.equals(p_element) || element_9.equals(p_element) || element_10.equals(p_element);
    }


    public T getElement(int index)
    {
                if (index == 1 )
            return element_1;
         else                 if (index == 2 )
            return element_2;
         else                 if (index == 3 )
            return element_3;
         else                 if (index == 4 )
            return element_4;
         else                 if (index == 5 )
            return element_5;
         else                 if (index == 6 )
            return element_6;
         else                 if (index == 7 )
            return element_7;
         else                 if (index == 8 )
            return element_8;
         else                 if (index == 9 )
            return element_9;
         else                 if (index == 10 )
            return element_10;
                        else
            return null;
    }

    @Override
    public InnerSet<T> addAllElements(Collection<? extends T> p_newC) {
        return null;
    }

    @Override
    public InnerSet<T> removeAllElements(Collection<?> p_cToDelete) {
        return null;
    }

    @Override
    public InnerSet<T> retainAllElements(Collection<?> p_cToKeep) {
        return null;
    }

    @Override
    public boolean containsAllElements(Collection<?> p_c) {
        return false;
    }

    @Override
    public String toString()
    {
        return "{" + element_1 + ", " + element_2 + ", " + element_3 + ", " + element_4 + ", " + element_5 + ", " + element_6 + ", " + element_7 + ", " + element_8 + ", " + element_9 + ", " + element_10 + " }";
    }
    
    @Override
    public int getSize()
    {
        return 10;
    }

}

File: src/scala/implementation/InnerSet_10.scala
package scala.implementation

object InnerSet_10
{
    def apply[T](element1: T, element2: T, element3: T, element4: T, element5: T, element6: T, element7: T, element8: T, element9: T, element10: T): InnerSet_10[T] = new InnerSet_10[T](element1, element2, element3, element4, element5, element6, element7, element8, element9, element10)
}

class InnerSet_10[T](element1: T, element2: T, element3: T, element4: T, element5: T, element6: T, element7: T, element8: T, element9: T, element10: T) extends InnerSet[T]
{

    override def addElement(p_element: T): InnerSet[T] =
    {
        if (element1.equals(p_element) || element2.equals(p_element) || element3.equals(p_element) || element4.equals(p_element) || element5.equals(p_element) || element6.equals(p_element) || element7.equals(p_element) || element8.equals(p_element) || element9.equals(p_element) || element10.equals(p_element))
        {
            this
        }
        else InnerHashSet(this)
    }

    override def addUnchecked(p_element: T): InnerSet[T] = InnerSet_11(element1, element2, element3, element4, element5, element6, element7, element8, element9, element10, p_element)


    override def addAllElements(p_innerSet: InnerSet[T]): InnerSet[T] =     {
                p_innerSet.addElement(element1)
                p_innerSet.addElement(element2)
                p_innerSet.addElement(element3)
                p_innerSet.addElement(element4)
                p_innerSet.addElement(element5)
                p_innerSet.addElement(element6)
                p_innerSet.addElement(element7)
                p_innerSet.addElement(element8)
                p_innerSet.addElement(element9)
                p_innerSet.addElement(element10)
            }
    override def containsElements(p_element: T): Boolean = element1.equals(p_element) || element2.equals(p_element) || element3.equals(p_element) || element4.equals(p_element) || element5.equals(p_element) || element6.equals(p_element) || element7.equals(p_element) || element8.equals(p_element) || element9.equals(p_element) || element10.equals(p_element)

    override def containsAllElements(p_innerSet: InnerSet[T]): Boolean =  p_innerSet.containsElements(element1) && p_innerSet.containsElements(element2) && p_innerSet.containsElements(element3) && p_innerSet.containsElements(element4) && p_innerSet.containsElements(element5) && p_innerSet.containsElements(element6) && p_innerSet.containsElements(element7) && p_innerSet.containsElements(element8) && p_innerSet.containsElements(element9) && p_innerSet.containsElements(element10)

    override def getElement(p_index: Int): Option[T] = p_index match{
        case 1 => Option(element1)
        case 2 => Option(element2)
        case 3 => Option(element3)
        case 4 => Option(element4)
        case 5 => Option(element5)
        case 6 => Option(element6)
        case 7 => Option(element7)
        case 8 => Option(element8)
        case 9 => Option(element9)
        case 10 => Option(element10)
        case _ => None
    }


    override def removeElement(p_element: T): InnerSet[T] =
    {
            if(element1.equals(p_element))
         new InnerSet_9(element2, element3, element4, element5, element6, element7, element8, element9, element10)
                  else             if(element2.equals(p_element))
         new InnerSet_9(element1, element3, element4, element5, element6, element7, element8, element9, element10)
                  else             if(element3.equals(p_element))
         new InnerSet_9(element1, element2, element4, element5, element6, element7, element8, element9, element10)
                  else             if(element4.equals(p_element))
         new InnerSet_9(element1, element2, element3, element5, element6, element7, element8, element9, element10)
                  else             if(element5.equals(p_element))
         new InnerSet_9(element1, element2, element3, element4, element6, element7, element8, element9, element10)
                  else             if(element6.equals(p_element))
         new InnerSet_9(element1, element2, element3, element4, element5, element7, element8, element9, element10)
                  else             if(element7.equals(p_element))
         new InnerSet_9(element1, element2, element3, element4, element5, element6, element8, element9, element10)
                  else             if(element8.equals(p_element))
         new InnerSet_9(element1, element2, element3, element4, element5, element6, element7, element9, element10)
                  else             if(element9.equals(p_element))
         new InnerSet_9(element1, element2, element3, element4, element5, element6, element7, element8, element10)
                  else             if(element10.equals(p_element))
         new InnerSet_9(element1, element2, element3, element4, element5, element6, element7, element8, element9)
                            else this
    }


    override def removeAllElements(p_innerSet: InnerSet[T]): InnerSet[T] =     {
                p_innerSet.removeElement(element1)
                p_innerSet.removeElement(element2)
                p_innerSet.removeElement(element3)
                p_innerSet.removeElement(element4)
                p_innerSet.removeElement(element5)
                p_innerSet.removeElement(element6)
                p_innerSet.removeElement(element7)
                p_innerSet.removeElement(element8)
                p_innerSet.removeElement(element9)
                p_innerSet.removeElement(element10)
            }

    override def retainAllElements(p_innerSet: InnerSet[T]): InnerSet[T] =
    {
      var i: InnerSet[T] = this
            if(!p_innerSet.containsElements(element1)) i = i.removeElement(element1)
            if(!p_innerSet.containsElements(element2)) i = i.removeElement(element2)
            if(!p_innerSet.containsElements(element3)) i = i.removeElement(element3)
            if(!p_innerSet.containsElements(element4)) i = i.removeElement(element4)
            if(!p_innerSet.containsElements(element5)) i = i.removeElement(element5)
            if(!p_innerSet.containsElements(element6)) i = i.removeElement(element6)
            if(!p_innerSet.containsElements(element7)) i = i.removeElement(element7)
            if(!p_innerSet.containsElements(element8)) i = i.removeElement(element8)
            if(!p_innerSet.containsElements(element9)) i = i.removeElement(element9)
            if(!p_innerSet.containsElements(element10)) i = i.removeElement(element10)
            i
    }

    override def getSize: Int = 10

    override def iterator: InnerSetIterator[T] = new InnerSetIterator(this)

    override def copy: Option[InnerSet[T]] = Option(new InnerSet_10(element1: T, element2: T, element3: T, element4: T, element5: T, element6: T, element7: T, element8: T, element9: T, element10: T))

    override def clear(unused: Boolean): InnerSet[T] = new InnerSet_0
}